### DISCUSSION TITLE
Getting Config Error while trying to install Oh My Posh

### DISCUSSION THREAD

**Discussion by jollygreen68** [JanDeDobbeleer/oh-my-posh#5616] [Troubleshoot]
Hey everyone, Linux noob here. Running Kubuntu 24.04.1 LTS that has Plasma 5.27. I saw this cool video last night about changing your bash prompt? I went to this site OhMyPosh and they have all these cool custom prompts already made that you can install. And I'd really like to install and use some of these great looking custom theme bash prompts. 

Problem is, I can't even get OhMyPosh installed properly and doing what I need to do, to even get the custom prompt installed. Here's the [video.](https://www.youtube.com/watch?v=nGHgyPLi7UM). I also tried following along from the OhMyPosh site, going step by step through there along with the video (the website is much harder to understand what to do as a beginner, the video helps a lot more). But I keep getting a Config Error bash prompt like this:
![Screenshot_20240914_172150](https://github.com/user-attachments/assets/47c6fc3b-28bc-4202-ad98-cd5713c5a8ee)

So I follow the video along to the letter T, and everything is fine until around min 2:19. Then it all falls apart, just like everything I try to do in Linux. I feel like Linux is some sort of way to torture you as a human being. Almost nothing ever works in Linux or easily. It's VERY frustrating. But I'm determined to make this work. 

So when I type in: vim ~/.bash_profile, everything goes to hell and a hand basket. It spits all this crazy stuff out:

E325: ATTENTION

Found a swap file by the name "~/.bash_profile.swp"

owned by: ryan dated: Sat Sep 14 12:36:40 2024

file name: ~ryan/.bash_profile

modified: YES

user name: ryan host name: SnowTrooper

process ID: 59892

While opening file "/home/ryan/.bash_profile"

CANNOT BE FOUND

(1) Another program may be editing the same file. If this is the case,

be careful not to end up with two different instances of the same

file when making changes. Quit, or continue with caution.

(2) An edit session for this file crashed.

If this is the case, use ":recover" or "vim -r /home/ryan/.bash_profile

"

to recover the changes (see ":help recovery").

If you did this already, delete the swap file "/home/ryan/.bash_profile

.swp"

to avoid this message.

Swap file "~/.bash_profile.swp" already exists!

[O]pen Read-Only, (E)dit anyway, (R)ecover, (D)elete it, (Q)uit, (A)bort:

So I went into my Home folder and found these 3 files that looked like bash_profile files and deleted them to start over so to speak. Because the above jargon made it sound like there were extra ones and you may need to delete one or all of them to continue. Also, in the video, when the person says to type in: export PATH=$PATH:/home/ubuntu/bin, then he "saves" his work. Yet HOW does he do that? I can't do that. There's no save button anywhere in my terminal. Nor can I figure out how to save. And then magically after he "saves his work", his terminal just all of a sudden changes back to a bash prompt that says: vim ~/.bash_profile

These were the 3 bash_profile files I found in my home folder that I deleted: 
.bash_profile.swm
.bash_profile.swo
.bash_profile.swp

Also, if I keep going past this part I get stuck on, it "seems" like it's working, but then always shows the CONFIG ERROR bash prompt once I enter in the commands: eval "$(oh-my-posh init bash --config /home/kubuntu/posh-themes/themes/dracula.omp.json)"

I just don't understand how I'm supposed to install this and since I barely understand Linux, it's making it crazy hard. It's like trying to figure out how to solve a very hard problem with some foreign language as the instructions. I just don't understand what to do unless the instructions tell you EXACTLY what to do. There is SO much left out, because people making the instructions just assume you know what to do. But a beginner doesn't. So I just need help figuring out how to get this to work please. I'm trying my best and google searching help and it's just not working. Thank you

**Comment by JanDeDobbeleer** [comment]
@jollygreen68 config error means that the file you reference in the `--config` flag either doesn't exist, or isn't accessible for your current user.

**Reply by jollygreen68** [reply]
Sure, yeah, I know what that means, haha. Like I mentioned right off the bat, I am a total beginner to Linux and computers. You will have to explain EXACTLY what that means and EXACTLY how I fix it, because I don't know what --config flag means. Thank you

**Reply by JanDeDobbeleer** [reply]
You added `eval "$(oh-my-posh init bash --config /home/kubuntu/posh-themes/themes/dracula.omp.json)"` to your `bashrc`. The error means that the file `/home/kubuntu/posh-themes/themes/dracula.omp.json` doesn't exist or can't be opened by your user (access rights).

I do advise to read up on linux knowledge prior to getting started with customization. It's not as out of the box as other operating systems.

**Reply by jollygreen68** [reply]
Yeah, that would take years of reading and learning just to understand enough to know what you're talking about. I'm not a computer programmer or expert level user. I just want my DE to look amazing. And your bash prompt themes look amazing. All I need is some simple help. 

So good news, I deleted everything, started over, and SOMEHOW I got the theme I wanted to work, to work! But, the only problem at this point is, HOW TO SAVE IT. Do you know how I can save it? I don't understand how to use VIM.

**Reply by jollygreen68** [reply]
OMG, this is such a nightmare. So I figured out, that when this guy was typing in this line of code in VIM: export PATH=$PATH:/home/ubuntu/bin as well as this command: eval "$(oh-my-posh init bash --config /home/ubuntu/posh-themes/themes/markbull.omp.json)" he was using the word ubuntu, because he's on a Windows OS computer and running a bash terminal. So I realized I needed to change that, so by luck and happenstance, I found out I need to change that Ubuntu part to Ryan. 

So now in VIM it's this: export PATH=$PATH:/home/ryan/bin

POSH_THEME="markbull"

eval "$(oh-my-posh init bash --config /home/ryan/posh-themes/themes/$POSH_THEME.omp.json)"

And it created the VIM bash_profile! And it showed it as working. But then, I closed out Terminal/Konsole, and the theme I chose is gone!?!? WTH. This is what happens when I type in ~/.bash_profile or $HOME/.bash_profile. Neither one works or opens the terminal theme I just installed?! 
ryan@SnowTrooper:~$ ~/.bash_profile
bash: /home/ryan/.bash_profile: Permission denied
ryan@SnowTrooper:~$ $HOME/.bash_profile
bash: /home/ryan/.bash_profile: Permission denied

Again, I'm SOOO close, can you or someone please just tell me how I get this to work. I've been working on this for two days straight and reading and researching and thinking critically to fix it. I'm RIGHT there, but just can't get it. Is there some setting in KDE Plasma that I have to change to make this new Oh My Posh Theme prompt show up?

**Reply by JanDeDobbeleer** [reply]
@jollygreen68 first off, if you're a novice to Linux, do not use VIM. It will indeed be a nightmare. [Install vscode](https://www.omgubuntu.co.uk/2021/06/how-to-install-visual-studio-code-on-ubuntu-20-04) and [set it as the EDITOR](https://code.visualstudio.com/docs/setup/linux#_setting-vs-code-as-the-default-text-editor). Restart your terminal afterwards and now you can open your `.bashrc` or `bash_profile` using vscode and edit it. 

All of your questions are Linux questions and have nothing to do with oh-my-posh. I don't mind you saying `that would take years of reading and learning` as long as you're aware that the people helping out here actually did do that. If you want to use Linux, you'll have to put in the work. There's no free lunch, these are basic Linux concepts you're struggling with. The issue above is a [file permission](https://www.freecodecamp.org/news/linux-chmod-chown-change-file-permissions/) issue for example, your user can't access that file as you most likely used `root` to create it. This is also highly likely why you got the config error as well. You can change the ownership to your user using `sudo chown ryan:ryan /home/ryan/.bash_profile`.

**Reply by jollygreen68** [reply]
Oh, my mistake. Is this not the place we ask for help? I see a ton of other people asking for help and getting responses here. Am I in the wrong place?  

Anyways, I just went back to my normal Linux forum I use and a really nice guy that has helped me before told me exactly what I was missing. It was one tiny little step, that is nowhere in the instructions on Oh My Posh website or on the guys video. It was super simple. And the cool thing is, I learned a lot just by him telling me how and why I needed to do that. Basically all I had to do, was put the command: eval "$(oh-my-posh init bash --config /home/ryan/posh-themes/markbull.omp.json)" inside my .BASHRC file at the bottom, save it, and Bob is Your Uncle! 

I did everything else on my own, with no help from anyone. Just sheer hard work, critical thinking and tons of watching the video over and over and reading the Oh My Posh tutorial instructions over and over. It was a nightmare. It's like trying to learn how to speak Spanish with instructions in Chinese. SO MUCH stuff is left out. It's like you're supposed to guess how to do a lot of this stuff. But, his info made a lot of sense in the grand scheme of things for the whole entire installation of Oh My Posh themes. I started to understand WHY I was doing certain things, WHY I needed to put this here or put that there, I learned a little bit about how to use VIM, so that I could follow along with the video, etc. But in the end, all I needed was a tiny bit of helpful info to make it work and save it. 

I do appreciate you "trying" to help. And I do appreciate you creating all of this, I know it probably took a LONG time. But I'm assuming it's a passion of yours? And that you use a terminal everyday all the time? I on the other hand do not. Nor do I need to or want to. Computers, programing and learning how to use Linux at an expert level are not things I want or need to learn. All I want to do is install programs I want to make my DE look pleasing and function well. So I don't need to learn all the ins and outs of Linux just so I can have a cool little terminal prompt. All I wanted was some simple help with a tiny little colored prompt. 

Not sure why computer people in general are SO against helping others? So far in my short time using Linux, 75% or more of the people that respond to posts are just rude, mean and totally unhelpful. It's like the computer community doesn't want to help anyone else. They're all angry and can't stand helping others. They just want to keep all the info to themselves or something. It's very strange. They also apparently have never heard the phrase: PAY IT BACK. Meaning: help others that know less once you become an expert. I practice that in my life in everything I do. I'm never too busy to help someone that has questions or needs to be shown how to do something I'm really good at. It's a pleasure for me to help others become better humans. Not sure why that's such a rarity in the computer world ?

**Reply by JanDeDobbeleer** [reply]
@jollygreen68 I'm not against helping others, otherwise I wouldn't be answering with links and pointing you in the right direction based on the information you provided.

Oh-my-posh is meant to be accessible, but I can't afford to be a personal debugger as I'm doing PR reviews, issue support and everything else in between. I have to draw a line somewhere. I did my best to read every message you wrote and reply with pointers. I'm glad you found someone who could be more helpful.
